scipy/special/tests/test_basic.py

```diff
@@ -1491,6 +1491,21 @@ def test_gh21426(self, dtype):
         result = special.betaincinv(a, a, x)
         assert_allclose(result, x, rtol=10 * np.finfo(dtype).eps)
 
+    @pytest.mark.parametrize("dtype, rtol",
+                             [(np.float32, 1e-4),
+                              (np.float64, 1e-15)])
```

Comment:
For single precision, the test passed only with `1000 * eps`, so I decided to manually set tolerances instead.